### PR TITLE
feat: Added Termii webhook parser

### PR DESCRIPTION
**Pull request by abhilipsasahoo03**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Added Webhook parser for Termii provider
- **Why was this change needed?** (You can also link to an open issue here)
Fixes  issue #1647 

Here is a link to resources that I referred to: https://developers.termii.com/events-and-reports

### CODE REVIEW

**Comment by abhilipsasahoo03** [comment]
@davidsoderberg @scopsy @davidsoderberg please review

File: packages/stateless/src/lib/provider/provider.interface.ts [hunk 1]
```diff
@@ -92,6 +92,10 @@ export enum SmsEventStatusEnum {
   FAILED = 'failed',
   UNDELIVERED = 'undelivered',
   RECEIVING = 'receiving',
+  DND_ACTIVE = 'DND activated',
```
Comment (p-fernandez) [inline packages/stateless/src/lib/provider/provider.interface.ts:95]:
```suggestion
  DND_ACTIVATED = 'DND activated',
```

File: providers/termii/src/lib/termii.provider.ts [hunk 2]
```diff
@@ -58,4 +60,60 @@ export class TermiiSmsProvider implements ISmsProvider {
...
+        return SmsEventStatusEnum.FAILED;
+      case 'Delivered':
+        return SmsEventStatusEnum.DELIVERED;
+      case 'Disconnected':
+        return SmsEventStatusEnum.DISCONNECTED;
+      case 'Rejected':
+        return SmsEventStatusEnum.REJECTED;
+      case 'Received':
+        return SmsEventStatusEnum.RECEIVED;
+    }
```
Comment (p-fernandez) [inline providers/termii/src/lib/termii.provider.ts:117]: @djabarovgeorge after what we discussed the other day which approach should we recommend to the contributor? I still find having hardcoded cases to be very brittle but that would mean this provider package wouldn't be agnostic to Novu.
@scopsy @davidsoderberg @ainouzgali this is a topic worth to be discussed inside the team.

Comment (djabarovgeorge) [reply providers/termii/src/lib/termii.provider.ts]: I would like to hear others opinion on the matter.

**Comment by abhilipsasahoo03** [comment]
@p-fernandez I've made the required changes. Please take a look.

**Comment by scopsy** [comment]
Let's wait to see if the assignee from the issue already started to work on this, if he did I will prefer to close this PR. Since it's not fair for the person who was assigned originally for the issue 🙏

**Comment by abhilipsasahoo03** [comment]
> Let's wait to see if the assignee from the issue already started to work on this, if he did I will prefer to close this PR. Since it's not fair for the person who was assigned originally for the issue 🙏

Extremely sorry. I had actually completely missed out on the assignment while creating the PR.

**Comment by abhilipsasahoo03** [comment]
Hi @scopsy kindly confirm and let me know if the assignee of this issue is willing to work on this issue or can this PR be merged as it is? You can also review the PR and suggest if anything needs to be modified. :)

**Comment by abhilipsasahoo03** [comment]
@scopsy please take a look

**Comment by abhilipsasahoo03** [comment]
@scopsy @davidsoderberg kindly review!

**Comment by abhilipsasahoo03** [comment]
@p-fernandez please take a look!

**Comment by abhilipsasahoo03** [comment]
Hi @scopsy @p-fernandez is it okay if we merge it?

**Comment by abhilipsasahoo03** [comment]
@p-fernandez @scopsy kindly review!

File: providers/termii/src/lib/termii.provider.ts [hunk 3]
```diff
@@ -58,4 +60,60 @@ export class TermiiSmsProvider implements ISmsProvider {
...
+
+    const status = this.getStatus(body.status);
+
+    if (status === undefined) {
+      return undefined;
+    }
+
+    return {
+      status: status,
+      date: new Date().toISOString(),
```
Comment (p-fernandez) [inline providers/termii/src/lib/termii.provider.ts:91]: Isn't the response returned from Termii providing any date?

File: packages/stateless/src/lib/provider/provider.interface.ts [hunk 4]
```diff
@@ -88,8 +88,16 @@ export enum SmsEventStatusEnum {
   SENT = 'sent',
   FAILED = 'failed',
   UNDELIVERED = 'undelivered',
+  RECEIVING = 'receiving',
```
Comment (p-fernandez) [inline packages/stateless/src/lib/provider/provider.interface.ts:91]: This is not used or needed.

File: packages/stateless/src/lib/provider/provider.interface.ts [hunk 5]
```diff
@@ -88,8 +88,16 @@ export enum SmsEventStatusEnum {
...
+  DELIVERY_UNCONFIRMED = 'delivery unconfirmed',
```
Comment (p-fernandez) [inline packages/stateless/src/lib/provider/provider.interface.ts:92]: This is not used or needed.

File: providers/termii/src/lib/termii.provider.ts [hunk 6]
```diff
@@ -58,4 +60,60 @@ export class TermiiSmsProvider implements ISmsProvider {
...
+        return SmsEventStatusEnum.DND_ACTIVATED;
+      case 'Expired':
+        return SmsEventStatusEnum.EXPIRED;
+      case 'Message sent':
+        return SmsEventStatusEnum.SENT;
+      case 'Message failed':
+        return SmsEventStatusEnum.FAILED;
+      case 'Delivered':
+        return SmsEventStatusEnum.DELIVERED;
+      case 'Disconnected':
```
Comment (p-fernandez) [inline providers/termii/src/lib/termii.provider.ts:111]: What's this status related to? What's disconnected in Termii?

**Review by p-fernandez — CHANGES_REQUESTED** [review]
Also adding some tests would be very helpful. You have some hints on what to test in Sendingblue provider.

**Pull request merged**